pkg/ucp/frontend/controller/resourcegroups/util.go

```diff
@@ -59,38 +61,161 @@ func (e *InvalidError) Is(err error) bool {
 	return ok
 }
 
-// ValidateDownstream can be used to find and validate the downstream URL for a resource.
-// Returns NotFoundError for the case where the plane or resource group does not exist.
-// Returns InvalidError for cases where the data is invalid, like when the resource provider is not configured.
-func ValidateDownstream(ctx context.Context, client store.StorageClient, id resources.ID) (*url.URL, error) {
+// ValidateRadiusPlane validates that the plane specified in the id exists. Returns NotFoundError if the plane does not exist.
```

Comment:
This code was refactored into a legacy path and a new UDT-aware path. I'd suggest ignoring the diff and just reviewing as new code.